src/js/modifiers.js

```diff
@@ -363,120 +363,114 @@ function modify_key_transpose() {
-        var index;
-
-        for (let i = 0; i < menulength; i++)
-        {
-            index = (semiconvergents) ? i : current_approximations.convergent_indicies[i];
-            
-            var n = parseInt(current_approximations.numerators[index]);
-            var d = parseInt(current_approximations.denominators[index]);
-            var prime_limit = current_approximations.ratio_limits[index];
-
-            var fraction_str = current_approximations.ratios[index];
-            var fraction = n / d;
-            
-            var cents_deviation = decimal_to_cents(fraction) - decimal_to_cents(interval);
-            var centsdabs = Math.abs(cents_deviation);
-            var cents_rounded = Math.round(10e6 * cents_deviation) / 10e6;
-
-            var centsdsgn;
-            if (cents_deviation / centsdabs >= 0)
-                centsdsgn = "+";
-            else
-                centsdsgn = "";
-        
-            var description = fraction_str+ " | " + centsdsgn + cents_rounded.toString() + "c | " + prime_limit + "-limit";
-
-            if (!interval) {
-                jQuery("#approximation_selection").append("<option selected disabled>Error: Invalid interval</option>");
-                break;
-            } else if (interval === fraction && interval) {  // for cases like 1200.0 === 2/1
-                jQuery("#approximation_selection").append("<option>"+description+"</option>");
-                break;
-            } else if ((centsdabs >= mincentsd && centsdabs <= maxcentsd) && (prime_limit >= minprime && prime_limit <= maxprime)) {
-                jQuery("#approximation_selection").append("<option>"+description+"</option>");
-            } else {
-                //debug("Option excluded: " + description);
-            }
-        }
-                
-        if (document.getElementById("approximation_selection").options.length === 0) {
-            semiconvergents ?
-                jQuery("#approximation_selection").append("<option selected disabled> None found, try to raise error tolerances.</option>") :
-                jQuery("#approximation_selection").append("<option selected disabled> Try to  \"Show next best approximations\" or edit filters.</option>")
-        }
+  jQuery("#approximation_selection").empty();
+
+  if (!(isEmpty(current_approximations))) {
+    var interval = line_to_decimal( jQuery ("#input_interval_to_approximate").val() );
+    var mincentsd = parseFloat( jQuery ( "#input_min_error").val() );
+    var maxcentsd = parseFloat( jQuery ( "#input_max_error").val() );
+    var minprime = parseInt( jQuery (" #input_approx_min_prime").val() );
+    var maxprime = parseInt( jQuery (" #input_approx_max_prime").val() );
+    var semiconvergents = !document.getElementById("input_show_convergents").checked;
+
+    if (minprime < 2) {
+      minprime = 2;
+      jQuery("#input_approx_min_prime").val(2);
+    }
+
+    if (maxprime > 7919) {
```

Comment:
To prevent future questions like `Why 7919? Where did that number come from?` I suggest reading that from the PRIMES array or creating a constant with that number.